### PR TITLE
Change getPathResult to return Result Path Error

### PR DESCRIPTION
**Pull request by jackkoenig**
It originally returned Result String Error which is problematic for
propagating Paths

Weirdly,  @john-sifive doesn't show up for assigning reviewers

### CODE REVIEW

File: share/wake/lib/system/path.wake [hunk 1]
```diff
@@ -17,10 +17,10 @@
 global def read path =
   def imp p = prim "read"
   match (getPathResult path)
-    Pass f = match (imp f)
+    Pass f = match (imp f.getPathName)
       Pass body = Pass body
       Fail f = Fail (makeError f)
-    x = x
+    Fail e = Fail e
```
Comment (jackkoenig) [inline share/wake/lib/system/path.wake]: This one was fun, I figured it out quickly but adding the `.getPathName` had no effect on the reported type error which was a bit surprising!

**Comment by terpstra** [comment]
I already have getPathError. Is that not appropriate in the case where you are using this modified getPathResult?

**Comment by jackkoenig** [comment]
@john-sifive needs `Result Path Error`, I guess he can do `(\x getPathError x | getOrPass x)`

**Comment by jackkoenig** [comment]
That's the spirit!

**Pull request merged**